CONTRIBUTING.md

```diff
@@ -35,9 +38,37 @@ To contribute to this repository:
 
           $ git push -u origin my-new-branch
 
-Finally, go to the web page of your fork of the `everware` repo,
+Finally, go to the web page of your fork of the `everware` repository,
 and click 'Pull request' to send your changes to the maintainers for
-review. This will send an email to the committers.
+review. This will send an email to the committers. A valid PR includes
+the following items:
+
+* a description of the problem you are solving or reference to existing issue
+* a discussion why you made the choices you made
+* contains the minimal set of changes needed to solve the problem (do
+  not mix several issues, rather create separate PRs)
+* unit and frontend tests for your changes
+* the code should follow the style of the project
+* the code should contain documentation in the form of docstrings
+  and comments
+* no commits merging master into your branch, instead rebase your branch
```

Comment:
in the beginning you write how one can create new branch using git CLI. maybe you can you add a link (like this: https://www.atlassian.com/git/tutorials/merging-vs-rebasing/) to rebase explanation (I'm not sure it's easy to put it here) and sentence of motivation why rebase instead of pull or squash?